cxx-qt-gen/src/syntax/cpp.rs

```diff
@@ -0,0 +1,10 @@
+// SPDX-FileCopyrightText: 2022 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+/// A generic C++ header and source pairing
+pub struct CppSourceCodePair {
```

Comment:
```suggestion
pub struct CppFragmentPair {
```
Sorry for bikeshedding this again :P This is kinda the same thing as `CppObject` in gen_cpp.rs, but it's not quite the same and easy to confuse them. Alternatively, we could have on struct for both of these use cases, but I think differentiating them is better.

Reply:
Come to think of it, why is this in the `syntax` submodule?

Reply:
Renamed to `CppFragmentPair` and moved into the generator module. I was thinking before it might be used in more places but it isn't so it can go there for now :-).